core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableProjectRule.java

```diff
@@ -42,6 +43,14 @@ protected EnumerableProjectRule(Config config) {
     super(config);
   }
 
+  @Override public boolean matches(RelOptRuleCall call) {
+    LogicalProject project = call.rel(0);
+    if (!project.getVariablesSet().isEmpty()) {
+      return false;
+    }
+    return true;
```

Comment:
Replace with `return project.getVariablesSet().isEmpty();` Idem to the other rules.